api/nodes/views.py

```diff
@@ -625,7 +625,7 @@ class NodeDraftRegistrationsList(JSONAPIBaseView, generics.ListCreateAPIView, No
     Use DraftRegistrationsList endpoint instead.
     """
     permission_classes = (
-        IsAdminContributor,
+        DraftRegistrationPermission,
```

Comment:
I can't comment on the actually relevant line:
My instinct is that the `get_queryset` should change to support the ticket requirements. Instead of listing all of the Drafts, this should list the Drafts on which the user is a contributor.

i.e., the queryset should return `user.draft_registractions_active.filter(branched_from=node)`
Existing behavior for the anonymous user would be to 401 -- which would redirect them to the login page -- but 

We should probably get some additional clarification from Product about a couple of things:
First: Should users who are contributors on the Draft but not the `branched_from` Project be able to see their Drafts on the list of the Project's Drafts (so long as the Project is public)?
Second: Should Project Admins be able to view all Drafts, even if they aren't contributors, or should that wait until the related ticket to give `branched_from` Project Admins implicit read permissions on Drafts even when they aren't contributors?
Third: What should the behavior be for Unauthenticated users? Should we 401, implicitly redirecting them to the login page? Or should we just return the Empty queryset?

I'm happy to own this communication if you would like.

Reply:
I discussed the questions you raised with Mark. 
1. If a user is a contributor to a draft registration but not to the main project, they should still be able to see their drafts if the project is public.
2. For now, we'll wait to implement a feature that gives project admins implicit read permissions on all related drafts.
3. Users who are not logged in should be redirected to the login page if they try to view draft registrations. If they don't have an account, they should be sent to a create an account page.

Reply:
Thanks for sorting this out!

All of that being true, there's a bit of work left to do. Mostly, this endpoint (`NodeDraftRegistrationList`) is currently returning the full list of DraftRegistrations to all Node contributors.

Two fixes to get the correct behavior:
First, you'll need to update this `get_queryset` function to return something like `user.active_draft_registrations.filter(branched_from=node)`. 

Second, you'll need to update your Permission class to use `can_view` instead of `has_permission(READ)` so that non-contributors on the Project can still see their associated Drafts.

Reply:
(I added some redundant-ish comments on tests to highlight where these requirements were being missed; we'll want/need to expand the test coverage, too)

Reply:
There are some unaddressed comments here regarding the queryset.

Reply:
Sorry, I had missed this comment. I have updated the PR.